src/MySqlConnector/Serialization/MySqlSession.cs

```diff
@@ -84,6 +84,8 @@ public void DoCancel(MySqlCommand commandToCancel, MySqlCommand killCommand)
 				// blocking the other thread for an extended duration.
 				killCommand.CommandTimeout = 3;
 				killCommand.ExecuteNonQuery();
+
+				commandToCancel.IsCanceled = true;
```

Comment:
I think this `IsCanceled` property should be combined with the existing `MySqlSession.m_state` so that all state related to cancellation is in one place.